cla_frontend/assets-src/javascripts/app/js/controllers/call_centre/eod_details_modal.js

```diff
@@ -0,0 +1,110 @@
+(function() {
+  'use strict';
+
+  angular.module('cla.controllers.operator')
+    .controller('EODDetailsModalCtrl',
+      ['$scope', 'case', 'eod_details', '$modalInstance', 'EXPRESSIONS_OF_DISSATISFACTION', 'EXPRESSIONS_OF_DISSATISFACTION_FLAGS', '$q', '$timeout', 'flash', 'postal',
+        function($scope, $case, eod_details, $modalInstance, EXPRESSIONS_OF_DISSATISFACTION, EXPRESSIONS_OF_DISSATISFACTION_FLAGS, $q, $timeout, flash, postal) {
+          $scope.case = $case;
+          $scope.EXPRESSIONS_OF_DISSATISFACTION = EXPRESSIONS_OF_DISSATISFACTION;
+
+          /*
+          $scope.getEODCategoryByValue = function(value) {
+            var matches = EXPRESSIONS_OF_DISSATISFACTION.filter(function(lookup) {
+              return lookup.value === value;
+            });
+            if(matches.length) {
+              return matches[0];
+            }
+          };
+          */
+
+          // focus on search field on open
+          $modalInstance.opened.then(function() {
+            $timeout(function() {
+              angular.element('#eod-modal-search').focus();
+            }, 50);
+          });
+
+          $scope.eod_details_model = {
+            categories: {},
+            notes: eod_details.notes || ''
+          };
+          angular.forEach(eod_details.categories, function(category) {
+            $scope.eod_details_model.categories[category.category] = {
+              is_major: category.is_major
+            };
+          });
+
+          $scope.isCategorySelected = function(category) {
+            return $scope.eod_details_model.categories[category] || false;
+          };
+
+          $scope.categoryNeedsMajorFlag = function(category) {
+            if(EXPRESSIONS_OF_DISSATISFACTION_FLAGS[category]) {
+              return EXPRESSIONS_OF_DISSATISFACTION_FLAGS[category].length > 1;
+            }
+            return null;
```

Comment:
the logic for `return null` was to mean that there is no such category; true/false otherwise